include/tracy.h

```diff
@@ -0,0 +1,105 @@
+enum thread_order : int32_t {
+	scheduler = 0x10000,               // scheduler thread (1)
+	executor = 0x10001,                // executor thread (1)
+	thread_queue = 0x10002,            // thread_queue instance (any number, no internal order)
+	immediate_lane = 0x10003,          // immediate pseudo-lane (1)
+	alloc_lane = 0x10004,              // alloc fiber (1)
+	host_first_lane = 0x10100,         // host lane (0..256)
+	first_device_first_lane = 0x10200, // first lane (0.256) of first device (0..256)
+	num_lanes_per_device = 0x100,
+	send_receive_first_lane = 0x21000, // first send/receive pseudo-lane (any number)
+};
+
+/// Tracy requires thread and fiber names to be live for the duration of the program, so if they are formatted dynamically, we need to leak them.
+inline const char* leak_name(const std::string& name) {
+	auto* leaked = malloc(name.size() + 1);
+	memcpy(leaked, name.data(), name.size() + 1);
+	return static_cast<const char*>(leaked);
+}
+
+} // namespace celerity::detail::tracy_detail
+
+#define CELERITY_DETAIL_IF_TRACY_SUPPORTED(...) __VA_ARGS__
+
+#else
+
+#define CELERITY_DETAIL_IF_TRACY_SUPPORTED(...)
+
+#endif
+
+
+#define CELERITY_DETAIL_IF_TRACY_ENABLED(...) CELERITY_DETAIL_IF_TRACY_SUPPORTED(if(::celerity::detail::tracy_detail::is_enabled()) { __VA_ARGS__; })
+#define CELERITY_DETAIL_IF_TRACY_ENABLED_FULL(...) CELERITY_DETAIL_IF_TRACY_SUPPORTED(if(::celerity::detail::tracy_detail::is_enabled_full()) { __VA_ARGS__; })
+
+#define CELERITY_DETAIL_TRACY_ZONE_SCOPED(TAG, COLOR_NAME)                                                                                                     \
+	CELERITY_DETAIL_IF_TRACY_SUPPORTED(ZoneNamedNC(___tracy_scoped_zone, TAG, ::tracy::Color::COLOR_NAME, ::celerity::detail::tracy_detail::is_enabled()))
+
+#define CELERITY_DETAIL_TRACY_ZONE_NAME(...)                                                                                                                   \
+	CELERITY_DETAIL_IF_TRACY_ENABLED_FULL(::celerity::detail::tracy_detail::apply_string([&](const auto& n) { ZoneName(n.data(), n.size()); }, __VA_ARGS__))
+#define CELERITY_DETAIL_TRACY_ZONE_TEXT(...)                                                                                                                   \
+	CELERITY_DETAIL_IF_TRACY_ENABLED_FULL(::celerity::detail::tracy_detail::apply_string([&](const auto& t) { ZoneText(t.data(), t.size()); }, __VA_ARGS__))
+
+#define CELERITY_DETAIL_TRACY_ZONE_SCOPED_V(TAG, COLOR_NAME, ...)                                                                                              \
+	CELERITY_DETAIL_TRACY_ZONE_SCOPED(TAG, COLOR_NAME);                                                                                                        \
+	CELERITY_DETAIL_TRACY_ZONE_NAME(__VA_ARGS__);
+
+#define CELERITY_DETAIL_TRACY_SET_THREAD_NAME_AND_ORDER(NAME, ORDER) CELERITY_DETAIL_IF_TRACY_ENABLED(::tracy::SetThreadNameWithHint(NAME, ORDER))
```

Comment:
Is this redundant with `celerity::detail::set_thread_name`? Should it just use this information, or should on macro do both (and the Tracy part only if there is Tracy)?

Reply:
`detail::set_thread_name` sets the POSIX thread name, whereas `TracySetThreadName` just updates the Tracy structures. Tracy in principle will detect the POSIX thread name, but I've observed this to be somewhat unreliable / racy.

Not sure what is to be gained from merging these since on the API side, `order` is also Tracy-specific.

Reply:
My idea regarding merging is that we want every thread that has a POSIX thread name to also have a Tracy thread name, and those names to be the same.

Reply:
I made a note to look into this later.